pkg/controller/controllercmd/cmd.go

```diff
@@ -97,44 +107,55 @@ func (c *ControllerCommandConfig) StartController(stopCh <-chan struct{}) error
 		return fmt.Errorf("unexpected config: %T", uncastConfig)
 	}
 
+	certDir := "/var/run/secrets/serving-certs"
+
 	observedFiles := []string{
 		c.basicFlags.ConfigFile,
+		// We observe these, so we they are created or modified by service serving cert signer, we can react and restart the process
+		// that will pick these up instead of generating the self-signed certs.
+		// NOTE: We are not observing the temporary, self-signed certificates.
+		filepath.Join(certDir, "tls.crt"),
+		filepath.Join(certDir, "tls.key"),
 	}
 
 	// if we don't have any serving cert/key pairs specified and the defaults are not present, generate a self-signed set
 	// TODO maybe this should be optional?  It's a little difficult to come up with a scenario where this is worse than nothing though.
 	if len(config.ServingInfo.CertFile) == 0 && len(config.ServingInfo.KeyFile) == 0 {
 		servingInfoCopy := config.ServingInfo.DeepCopy()
 		configdefaults.SetRecommendedHTTPServingInfoDefaults(servingInfoCopy)
-		_, keyErr := os.Stat(servingInfoCopy.KeyFile)
-		_, certErr := os.Stat(servingInfoCopy.CertFile)
-		if os.IsNotExist(keyErr) && os.IsNotExist(certErr) {
-			certDir, err := ioutil.TempDir("", "serving-cert-")
+
+		if hasServiceServingCerts(certDir) {
```

Comment:
do we have a race here if the files are mounted between here and the observer start?

Reply:
> do we have a race here if the files are mounted between here and the observer start?

I would take  TODO if it becomes a practical problem.

Reply:
yes, TODO is fine.